azure/services/scalesets/scalesets.go

```diff
@@ -564,11 +564,14 @@ func (s *Service) buildVMSSFromSpec(ctx context.Context, vmssSpec azure.ScaleSet
 			nicConfig := compute.VirtualMachineScaleSetNetworkConfiguration{}
 			nicConfig.VirtualMachineScaleSetNetworkConfigurationProperties = &compute.VirtualMachineScaleSetNetworkConfigurationProperties{}
 			nicConfig.Name = pointer.String(vmssSpec.Name + "-" + strconv.Itoa(i))
-			if pointer.BoolDeref(n.AcceleratedNetworking, false) {
-				nicConfig.VirtualMachineScaleSetNetworkConfigurationProperties.EnableAcceleratedNetworking = pointer.Bool(true)
+			nicConfig.EnableIPForwarding = pointer.Bool(true)
```

Comment:
why is this being harcoded to true?

Reply:
the default case when no `networkInterfaces` are configured, the forwarding is hardcoded:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3243/files#diff-a6779b0b37d0dcdbe7dcd6f00150a22e12ac067a96269038452d27ba39349a90R517

we could also set this just to the first interface, but I'm unsure about the implications. Possibly for Linux nodes it works anyway (though that might depend on the CNI) but for Windows nodes I'm not sure if it would be required for all interfaces or not. 
Also regarding the hardcoding: there's no property on the NetworkInterfaces to set this, and if we'd add it we'd probably need a validation webhook to ensure the first interface has set it to true.